File: algebra/gcd/clojure/gcd.clj
;;euclid-gcd : A function to find the gcd of 2 numbers using the euclidean algorithm
;;  a,b -> gcd (a,b)

(defn euclid-gcd [a b]
  "Returns the gcd of 2 non-negative numbers"
  (if (= b 0)
      a
      (euclid-gcd b (rem a b))))

;;Tests:

(assert (= 2 (euclid-gcd 0 2)))
(assert (= 1 (euclid-gcd 5 7)))
(assert (= 2 (euclid-gcd 4 6)))
(assert (= 10 (euclid-gcd 100 70)))

File: algebra/gcd/python/gcd.py
"""
    euclid_gcd(0 is used to find the gcd of two numbers using euclidean algorithm
    @input: a, b numbers
    @output gcd(a,b)
"""

def euclid_gcd(a, b):
    
    if a == 0:
        return b
    
    if b == 0 :
        return a
    
    if a < b:
        return euclid_gcd(b, a)

    while b != 0:
        (a, b) = (b, a%b)

    return a
